### ISSUE TITLE
twisted.internet.interfaces.ITransport.write() docstring should mention it takes a parameter of bytes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8636]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8636|
|Type|enhancement|
|Created|2016-07-20 12:05:24Z|
|Branch|https://github.com/twisted/twisted/tree/8636-rodrigc-itransport-doc|

twisted.internet.interfaces.ITransport.write()
and twisted.internet.interfaces.ITransport.writeSequence() should
mention that they bytes parameters, not str parameters

<details><summary>Searchable metadata</summary>

```
trac-id__8636 8636
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8636_rodrigc_itransport_doc 8636-rodrigc-itransport-doc
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469016324183424 1469016324183424
changetime__1469144720057342 1469144720057342
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/366

**Comment by twisted-trac** [comment]
|[<img alt="scop's avatar" src="https://avatars.githubusercontent.com/u/109152?s=50" width="50" height="50">](https://github.com/scop)<a name="note_3"></a>|@scop set owner to @rodrigc|
|-|-|

Looks good to me, please merge.

FWIW it also seems to me that it could be considered to relax the restriction of operating only on lists of bytes to working on any iterable of bytes. A quick peek at the in-tree implementations shows that none of them would mind. But I suppose that's something for another day.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset ad6a86f027e9c5a6d2edcd279626de83f6fe88a0
```
#!CommitTicketReference repository="" revision="ad6a86f027e9c5a6d2edcd279626de83f6fe88a0"
Merge pull request #366 from twisted/8636-rodrigc-itransport-doc

Author: rodrigc
Reviewer: scop
Fixes: #8636

Clarify docstring for ITransport.write() and ITransport.writeSequence()
```
